test/helpers/kubectl.go

```diff
@@ -3405,6 +3405,32 @@ func (kub *Kubectl) WaitForIPCacheEntry(node, ipAddr string) error {
 		&TimeoutConfig{Timeout: HelperTimeout})
 }
 
+// KeepRunning runs command on repeat in goroutine until channel
+// is closed and closes run channel when command is first run
+func (kub *Kubectl) KeepRunning(cmd string, quit, run chan struct{}) {
+	go func() {
+		firstRun := true
+	loop:
+		for {
+			select {
+			case <-quit:
+				break loop
+			default:
+				res := kub.Exec(cmd)
+				if !res.WasSuccessful() {
+					kub.Logger().WithFields(logrus.Fields{
+						"cmd": cmd,
+					}).Warning("Command failed running in the background")
+				}
+				if firstRun {
+					close(run)
+				}
+				firstRun = false
```

Comment:
Are you sure that you don't need a sleep or some pacing?

Reply:
from my tests `kill 1` instantly causes the pod to be `completed` and restart, I think this is what we want to have in this test. I we wait, we risk that Cilium will actually restart.

Reply:
As for pacing, I think that 2 sequential kubectl should be enough.